test/unittest/utils/SystemInfoTests.cpp

```diff
@@ -67,6 +76,44 @@ TEST(SystemInfoTests, GetUsernameTest)
     EXPECT_FALSE(username.empty());
 }
 
+/**
+ * This test checks the file_exists static method of the SystemInfo class
+ */
+TEST(SystemInfoTests, FileExistsTest)
+{
+    // 1. Check that a valid file in the current directory exists
+    std::string filename = "environment_test_file.json";
+    EXPECT_TRUE(eprosima::SystemInfo::file_exists(filename));
+
+    // 2. Check that a valid path and filename returns correctly
+    char buffer[SIZE];
+    char* current_dir = getcwd(buffer, SIZE);
+    if (current_dir)
+    {
+        filename = current_dir;
+        filename += "/environment_test_file.json";
+    }
```

Comment:
If (for whatever reason) we don't get `current_dir`, we will be checking with the same filename again. Making the test fail in that case would be more informative. It was done this way for the `non_existent.txt` file below.

Or, if we know that `getcwd` will fail for some platforms, we can include the `EXPECT` inside the if block and skip it completely on those platforms. Adding a comment stating why we are doing this would be a plus.

Reply:
I addressed this on commit c98fec6d9